internal/core/saptune/saptune.go

```diff
@@ -0,0 +1,120 @@
+package saptune
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/utils"
+	"golang.org/x/mod/semver"
+)
+
+var (
+	ErrSaptuneVersionUnknown = errors.New("could not determine saptune version")
+	ErrUnsupportedSaptuneVer = errors.New("saptune version is not supported")
+)
+
+const (
+	MinimalSaptuneVersion = "v3.1.0"
+)
+
+type Saptune struct {
```

Comment:
Most probably you want to revisit this to use the new format with `package_version`, `saptune_installed`, etc